File: portal/portal/App_Code/DAO/PaginaAcessoDAO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using MySql.Data.MySqlClient;
using System.IO;


namespace portal.App_Code.DAO
{
    public class PaginaAcessoDAO
    {
        //lendo string de conexão     
        String sc = Properties.Settings.Default.CN;
        String logPath = Properties.Settings.Default.PathErrorLog;

        public PaginaAcesso carregar(long pCodigo)
        {
            PaginaAcesso obj = new PaginaAcesso();
            try
            {
                //cria a conexao com o bd
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Pagina_Acesso where codigo=" + pCodigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                if (registros.Read())
                {
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.Nome = registros["nome"].ToString();
                    obj.Descritivo = registros["descritivo"].ToString();
                    obj.Pagina = registros["pagina"].ToString();
                    obj.Status = (PaginaAcesso.TipoStatus)registros["status"];
                }
                conexao.Close();
            }
            catch (Exception err)
            {
                String log = "Erro=>" + DateTime.Now + err.Message + Environment.NewLine;
                File.AppendAllText(logPath, log);
            }
            return obj;
        }

        public List<PaginaAcesso> carregarLista(String[] pFiltros, String pOrdena)
        {
            List<PaginaAcesso> lista = new List<PaginaAcesso>();
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Pagina_Acesso where ";
                for (int i = 0; i < pFiltros.Length; i++)
                {
                    sql = sql + pFiltros[i];
                }
                sql = sql + " order by " + pOrdena;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                while (registros.Read())
                {
                    PaginaAcesso obj = new PaginaAcesso();
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.Nome = registros["nome"].ToString();
                    obj.Descritivo = registros["descritivo"].ToString();
                    obj.Pagina = registros["pagina"].ToString();
                    obj.Status = (PaginaAcesso.TipoStatus)registros["status"];
                    lista.Add(obj);
                }
                conexao.Close();
            }
            catch (Exception err)
            {

            }
            return lista;
        }


        public static List<PaginaAcesso> carregarLista(double pCodigo)
        {
            String sc = Properties.Settings.Default.CN;
            String logPath = Properties.Settings.Default.PathErrorLog;

            List<PaginaAcesso> lista = new List<PaginaAcesso>();
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                String sql = "select t1.* from Pagina_Acesso t1,permissao_acesso t2 where t1.codigo_pagina=t2.codigo_grupo and t2.codigo_grupo="+ pCodigo;

                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                while (registros.Read())
                {
                    PaginaAcesso obj = new PaginaAcesso();
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.Nome = registros["nome"].ToString();
                    obj.Descritivo = registros["descritivo"].ToString();
                    obj.Pagina = registros["pagina"].ToString();
                    obj.Status = (PaginaAcesso.TipoStatus)registros["status"];
                    lista.Add(obj);
                }
                conexao.Close();
            }
            catch (Exception err)
            {

            }
            return lista;
        }

        public void persistir(PaginaAcesso obj)
        {
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "";
                if (obj.Codigo == 0)
                {
                    sql = "insert into Pagina_Acesso(nome, descritivo, pagina, status) value('{0}','{1}','{2}',{3})";
                    sql = String.Format(sql, obj.Nome, obj.Descritivo, obj.Pagina, obj.Status);
                }
                else
                {
                    sql = "update Pagina_Acesso set nome='{0}', descritivo='{1}', pagina='{2}', status={3} where codigo={4}";
                    sql = String.Format(sql, obj.Nome, obj.Descritivo, obj.Pagina, obj.Status, obj.Codigo);
                }

                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }


        public void remover(PaginaAcesso obj)
        {
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "delete from Pagina_Acesso where codigo=" + obj.Codigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }

    }
}

File: portal/portal/App_Code/DAO/PaginaAcessoDAO.php
<?php
namespace portal\App_Code\DAO {

//CHAMADAS EXTERNAS------------------------------------------------------------------------------------------------------------------------------
use portal\App_Code\PaginaAcesso;
mysqli_report(MYSQLI_REPORT_STRICT|MYSQLI_REPORT_ERROR);     

    try{
        require_once("/../PaginaAcesso.php");        
        require_once("/../../connection.php");        
    }catch(Exception $e){
		require_once (dirname(__file__, 2) . "\PaginaAcesso.php");
		require_once (dirname(__file__, 3) . "\connection.php");
    }

class PaginaAcessoDAO{
//INSERIR----------------------------------------------------------------------------------------------------------------------------------------
    function inserirPaginaAcesso($pg){
        try{
            $con = BD_AbrirConexao();
            $sql = "insert into pagina_acesso (nome, descritivo,pagina,status) values (' " .
                 $pg->getNome() . " ',' " .
                 $pg->getDescritivo() . " ',' " .
                 $pg->getPagina() . " ',' " .
                 $pg->getTipoStatus() . " ')";
                 mysqli_query($con,$sql);
				 echo "Inserção com sucesso!";		 
        }catch (Exception $e){
                echo "Erro de inserção: " . $e;
        }
    mysqli_close($con);	
	}

//ALTERAR--------------------------------------------------------------------------------------------------------------------------------------- 
function alterarPaginaAcesso($pg){  
        try{
            $con = BD_AbrirConexao();
            $sql = "update pagina_acesso set " . 
                "nome = ' " . $pg->getNome() . "', " .
                "descritivo = ' " . $pg->getDescritivo() . "', " .
                "pagina = ' " . $pg->getPagina() . "', " .
                "status = ' " . $pg->getTipoStatus() . "' " .
                "where codigo = " . $pg->getCodigo();
                mysqli_query($con,$sql);
				echo "Dados alterados com sucesso";
        }catch (Exception $e){
                echo "Erro ao alterar dados: " . $e;
                }
    mysqli_close($con);	
    }

//CARREGAR--------------------------------------------------------------------------------------------------------------------------------------
    function carregarPaginaAcesso ($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "select * from pagina_acesso where codigo = " . $codigo;
                $gargantula = mysqli_query($con,$sql);
                $pg = new PaginaAcesso();
                    if($carga = mysqli_fetch_array($gargantula)){
                        $pg->setPaginaAcesso(($carga["codigo"]),
											 ($carga["nome"]),
											 ($carga["descritivo"]),
											 ($carga["pagina"]),
											 ($carga["status"]));
						mysqli_close($con);
						return $pg;								 
                    }else{
                        echo "Registros não encontrados!";
                    }
            }catch (Exception $e){
                echo "Erro de busca: " . $e;
            }
	mysqli_close($con);			
    }

//CARREGAR LISTA DE PÁGINAS TIPO PaginaAcesso MODO 1-------------------------------------------------------------------------------------------
    function carregarListaPaginaAcesso($qtdOpcoes, $ordem){
        $sql = "select * from pagina_acesso where ";

        for ($i = 0 ; $i <= strlen((array)$qtdOpcoes) ; $i++){
            $sql = $sql . $qtdOpcoes[$i];
                if($i != $qtdOpcoes[$i] - 1){
                        $sql = $sql . "and ";  
                } 
        } 
        try{
            $con = BD_AbrirConexao();
            $gargantula = mysqli_query($con,$sql);
                    if($carga = mysqli_fetch_array($gargantula)){
                        $listaPaginaAcesso = [];
                        while($carga = mysqli_fetch_array($gargantula)){
                            $pg = new PaginaAcesso();
                            $pg->setPaginaAcesso(($carga["codigo"]),
											     ($carga["nome"]),
											     ($carga["descritivo"]),
											     ($carga["pagina"]),
											     ($carga["status"]));
                            $listaPaginaAcesso[] = $pg;
                        }
                        mysqli_close($con);
                        return $listaPaginaAcesso;
                    }else{
                        echo "Registros não encontrados!";
                    }
        }catch(Exception $e){
                echo "Erro de busca: " . $e;           
            }
mysqli_close($con);
    }
    
//CARREGAR LISTA DE PÁGINAS TIPO PaginaAcesso MODO 2-------------------------------------------------------------------------------------------
    function carregarListaPaginaAcessoCod($dado){
        try{
			$sql = "select t1.* from Pagina_Acesso t1,permissao_acesso t2 where t1.codigo_pagina = t2.codigo_grupo and t2.codigo_grupo= " . $dado;
            $con = BD_AbrirConexao();
            $gargantula = mysqli_query($con,$sql);
                if($carga = mysqli_fetch_array($gargantula)){
						  $listaPaginaAcesso = [];
                       while($carga = mysqli_fetch_array($gargantula)){
                             $pg = new PaginaAcesso();
                             $pg->setPaginaAcesso(($carga["codigo"]),
											     ($carga["nome"]),
											     ($carga["descritivo"]),
											     ($carga["pagina"]),
											     ($carga["status"]));
                            $listaPaginaAcesso[] = $pg;                
                        }
                    mysqli_close($con);
                    return $listaPaginaAcesso;                    
                }else {
                        echo "Registros não encontrados!";
                }
         }catch(Exception $e){
                echo "Erro ao carregar dados: " . $e;
         }
        mysqli_close($con);        
    }

//DELETAR----------------------------------------------------------------------------------------------------------------------------------------
    function deletarPaginaAcesso($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "delete from pagina_acesso where codigo = " . $codigo; 
                $carga = mysqli_query ($con,$sql);
				echo "Exclusão com sucesso";
            }catch(Exception $e){
                echo "Erro de exclusão: " . $e;
            }
		mysqli_close($con);
    }
  }
}
?>